orm/src/main/java/org/hibernate/search/jmx/impl/IndexControl.java

```diff
@@ -95,33 +96,27 @@ public void index(String entity) {
 			Thread.currentThread().interrupt();
 			throw new SearchException( "Unable to complete indexing" );
 		}
-		session.close();
 	}
 
 	@Override
 	public void optimize(String entity) {
 		Class<?> clazz = getEntityClass( entity );
-
 		SessionFactory factory = getSessionFactory();
-		Session session = factory.openSession();
-		FullTextSession fullTextSession = Search.getFullTextSession( session );
-		fullTextSession.beginTransaction();
-		fullTextSession.getSearchFactory().optimize( clazz );
-		fullTextSession.getTransaction().commit();
-		session.close();
+		ExtendedSearchIntegrator searchIntegrator = ContextHelper.getSearchIntegratorBySF( factory );
+		searchIntegrator.optimize( clazz );
 	}
 
 	@Override
 	public void purge(String entity) {
 		Class<?> clazz = getEntityClass( entity );
 
 		SessionFactory factory = getSessionFactory();
-		Session session = factory.openSession();
-		FullTextSession fullTextSession = Search.getFullTextSession( session );
-		fullTextSession.beginTransaction();
-		fullTextSession.purgeAll( clazz );
-		fullTextSession.getTransaction().commit();
-		session.close();
+		try ( Session session = factory.openSession() ) {
```

Comment:
I guess you mean because the `FullTextSession` does nothing more than delegating to `Session`? I hadn't noticed that.

So you're right, it is not *required* to close both. It shouldn't hurt (`Closeable#close` is expected to be idempotent as per its javadoc), but the current implementation makes it useless.

Well, never mind.